### PR TITLE
HPCC-14704 Assume that assertex() conditions are likely to be true

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@mark please review

**Comment by ghalliday** [comment]
This improved the timing an eclcc example by a small amount.  Maybe 1-2%

**Comment by mckellyln** [comment]
Sure, looks good.

File: system/jlib/jiface.hpp [hunk 1]
```diff
@@ -34,19 +34,19 @@ void jlib_decl raiseAssertCore(const char *assertion, const char *file, unsigned
 
 #ifdef _PREFAST_
  #pragma warning (disable : 6244)   // generates too much noise at the moment
- #define assertex(p) ((p) ? ((void) 0) : (( (void) raiseAssertException(#p, __FILE__, __LINE__), __analysis_assume(p))))
- #define assert(p) ((p) ? ((void) 0) : (( (void) raiseAssertCore(#p, __FILE__, __LINE__), __analysis_assume(p))))
+ #define assertex(p) (likely(p) ? ((void) 0) : (( (void) raiseAssertException(#p, __FILE__, __LINE__), __analysis_assume(p))))
```
Comment (richardkchapman) [inline system/jlib/jiface.hpp]: What does likely(p) do that __analysis_assume(p) did not?

Comment (ghalliday) [inline system/jlib/jiface.hpp]: I think __analysis_assume is a visual studio feature, primarily used to help a static analysis tool. (Used by PREfast)

**Pull request merged**